src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -249,10 +249,16 @@ void CslTxScheduler::HandleSentFrame(const Mac::TxFrame &aFrame, Error aError, C
         break;
 
     case kErrorNoAck:
-        aChild.IncrementCslTxAttempts();
+        if (aFrame.IsHeaderUpdated())
+        {
+            // Only consider a transmission attempt the case when the radio platform returns back a
+            // frame with an updated header, i.e.: there was not an abrupt transmission operation
+            // cancelation.
```

Comment:
The `kErrorNoAck` error indicates that the frame was sent but no ack was received (ack timeout).

I think it then make sense to assume/require that with this error, we would never have frame header not updated.
(if frame is sent over the air, then all fields should be set on it).

If the frame is not sent over the air at all, then we can have other error codes to indicate this:
- `kErrorChannelAccessFailure` to indicate we tried to access channel but channel was busy.
- `kErrorAbort` to indicate frame tx was abruptly canceled/aborted.

I think we just need to check and handle `IsHeaderUpdate()` under the these errors and undere `kErrorNoAcks` we can consider adding an assert ensuring `IsHeaderUpdated() == true`.

Reply:
Agree, added the assert.